### PR TITLE
[instruments] updated README and test plan for widgets

### PR DESCRIPTION
**Pull request by maltheism**
## Brief summary of changes

- [x] Have you updated related documentation?

#### Testing instructions (if applicable)

1. Read the README.md for the instruments module.
2. Follow the TestPlan.md instructions for the instruments module.

#### Link(s) to related issue(s)

* Resolves #  (Reference the issue this fixes, if any.)
#6437

### CODE REVIEW

File: modules/instruments/test/TestPlan.md [hunk 1]
```diff
@@ -43,3 +43,14 @@ instrument for that candidate. Enter a date (same a step 4) posterior to the DoD
...
+13. Dashboard Widget - "My Tasks" for Incomplete forms
+    - Ensure the total of Incomplete forms correspond with the correct 
+        number of candidates inside the statistics/statistics_site module.
+    - Click on the total number of Incomplete forms and check if redirection
+        to the statistics/statistics_site module succeeds.
+14. Candidate Widget - "Behavioural Data" for Candidate instruments entry.
+    - Ensure the appropriate visit labels are visible for the candidate. 
+    - The visit status, subproject status, site, date of visit, and age should be shown.
+    - Clicking on a visit should redirect to the instrument_list module for the candidate.
+    - Test making a new Time Point for a candidate and assign a Visit label. The Behavioural Data widget should update
```
Comment (PapillonMcGill) [inline modules/instruments/test/TestPlan.md:55]:
```suggestion
    - Test making a new Time Point for a candidate and assign an instrument to the time point. The Behavioural Data widget should update accordingly.
```

File: modules/instruments/test/TestPlan.md [hunk 2]
```diff
@@ -43,3 +43,14 @@ instrument for that candidate. Enter a date (same a step 4) posterior to the DoD
...
+    - Ensure the total of Incomplete forms correspond with the correct 
+        number of candidates inside the statistics/statistics_site module.
+    - Click on the total number of Incomplete forms and check if redirection
+        to the statistics/statistics_site module succeeds.
+14. Candidate Widget - "Behavioural Data" for Candidate instruments entry.
+    - Ensure the appropriate visit labels are visible for the candidate. 
+    - The visit status, subproject status, site, date of visit, and age should be shown.
+    - Clicking on a visit should redirect to the instrument_list module for the candidate.
+    - Test making a new Time Point for a candidate and assign a Visit label. The Behavioural Data widget should update 
+        to include the new Time Point by showing the visit label with the subproject, site, and status as not started.
```
Comment (PapillonMcGill) [inline modules/instruments/test/TestPlan.md:56]:
```suggestion
        to include the new Time Point by showing the instrument with the subproject, site, and status as not started.
```

**Comment by maltheism** [comment]
thanks for the help @PapillonMcGill !

File: modules/instruments/test/TestPlan.md (hunk 2 above)
Comment (PapillonMcGill) [reply modules/instruments/test/TestPlan.md]: I made the 2 previous comment with a length of time in between. This line is now superfluous.

**Comment by ridz1208** [comment]
@maltheism can you make sure there to remove the old sections in the dashboard testplans that are now in the instruments one please

File: modules/instruments/test/TestPlan.md [hunk 3]
```diff
@@ -43,3 +43,13 @@ instrument for that candidate. Enter a date (same a step 4) posterior to the DoD
 12. Verify Validity flags are functioning for instruments.
     - **Validity:** can be marked as “Valid”, “Questionable”, or “Invalid”. Whether or not this flag is shown for an instrument is by the boolean $ValidityEnabled. Whether the field is required before flagging an instrument as complete is determined by $ValidityRequired.
     - You can test the forgoing flags by switching the corresponding boolean of the test instruments to either `true` or `false`.
+13. Dashboard Widget - "My Tasks" for Incomplete forms
```
Comment (driusan) [inline modules/instruments/test/TestPlan.md:46]: Can you put these in different sections? It's not clear the way it's written that the user needs to go to a different module to validate these things.

Comment (maltheism) [reply modules/instruments/test/TestPlan.md]: @driusan done

**Comment by maltheism** [comment]
@ridz1208 @driusan added the requested changes.

**Comment by ridz1208** [comment]
@driusan final review

File: modules/instruments/README.md [hunk 4]
```diff
@@ -40,3 +40,10 @@ instruments, but does not load through this module.
 
 `bvl_feedback` threads for an instrument may be created on this
 module.
+
+## Dashboard Widget
```
Comment (driusan) [inline modules/instruments/README.md:44]:
```suggestion
### Dashboard Widget
```

The header level is wrong. The dashboard widget is an interaction of the module.

File: modules/instruments/test/TestPlan.md [hunk 5]
```diff
@@ -43,3 +43,18 @@ instrument for that candidate. Enter a date (same a step 4) posterior to the DoD
 12. Verify Validity flags are functioning for instruments.
     - **Validity:** can be marked as “Valid”, “Questionable”, or “Invalid”. Whether or not this flag is shown for an instrument is by the boolean $ValidityEnabled. Whether the field is required before flagging an instrument as complete is determined by $ValidityRequired.
     - You can test the forgoing flags by switching the corresponding boolean of the test instruments to either `true` or `false`.
+
+### The Dashboard module corresponding to Instruments module.
+   1. Dashboard Widget - "My Tasks" for Incomplete forms
```
Comment (driusan) [inline modules/instruments/test/TestPlan.md:48]: This should tell people they need to go to the dashboard to do this test.

File: modules/instruments/test/TestPlan.md [hunk 6]
```diff
@@ -43,3 +43,18 @@ instrument for that candidate. Enter a date (same a step 4) posterior to the DoD
...
+
+### The Dashboard module corresponding to Instruments module.
+   1. Dashboard Widget - "My Tasks" for Incomplete forms 
+      - Ensure the total of Incomplete forms correspond with the correct 
+            number of candidates inside the statistics/statistics_site module.
+      - Click on the total number of Incomplete forms and check if redirection
+            to the statistics/statistics_site module succeeds.
+
+### The Candidate_list module corresponding to Instruments module.
+   1. Candidate Widget - "Behavioural Data" for Candidate instruments entry.
```
Comment (driusan) [inline modules/instruments/test/TestPlan.md:55]: This should tell people that they need to go to the candidate dashboard for a candidate.

File: modules/instruments/README.md [hunk 7]
```diff
@@ -40,3 +40,10 @@ instruments, but does not load through this module.
...
+
+The total `Incomplete forms` will be shown inside the Dashboard widget, 
+named "My Tasks" and with a link redirecting to statistics/statistics_site. 
+The number of candidates with incomplete forms for the corresponding instruments will be listed. 
+You can click on a candidate to be redirected to the instruments module and complete the instrument form for the candidate.
```
Comment (driusan) [inline modules/instruments/README.md:49]: The last 2 sentences (paragraphs?) belong in the test plan, not the spec.

Comment (maltheism) [reply modules/instruments/README.md]: @driusan I removed them in latest commit and a rephrased version of the sentences already exist in the test plan.

File: modules/instruments/test/TestPlan.md (hunk 5 above)
Comment (maltheism) [reply modules/instruments/test/TestPlan.md:48]: @driusan fixed

File: modules/instruments/test/TestPlan.md (hunk 6 above)
Comment (maltheism) [reply modules/instruments/test/TestPlan.md:55]: @driusan fixed.

File: modules/instruments/test/TestPlan.md (hunk 5 above)
Comment (driusan) [reply modules/instruments/test/TestPlan.md:48]: I don't see the fix.. did you forget to push it?

**Pull request merged**